File: java/src/test/java/StrategyTest.java
import org.junit.jupiter.api.Test;
import strategy.AbcBankRemit;
import strategy.DefBankRemit;
import strategy.Remit;
import strategy.fee.DefaultFee;
import strategy.fee.Fee;
import strategy.fee.PartnershipFee;
import static org.junit.jupiter.api.Assertions.*;

public class StrategyTest {

    @Test
    void StrategyRun() {
        var amount = 1000;
        var defaultFee =  1000 * Fee.DEFAULT_FEE;
        var partnershipFee = 1000 * Fee.PARTNERSHIP_FEE;

        Remit abcBankRemit = new AbcBankRemit(new DefaultFee());
        Remit defBankRemit = new DefBankRemit(new PartnershipFee());

        var amountAbc = abcBankRemit.calculateAmount(amount);
        var amountDef = defBankRemit.calculateAmount(amount);

        assertEquals(amount + defaultFee, amountAbc);
        assertEquals(amount + partnershipFee, amountDef);
    }
}

File: kotlin/src/test/kotlin/StrategyTest.kt
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import strategy.AbcBankRemit
import strategy.DefBankRemit
import strategy.fee.DefaultFee
import strategy.fee.Fee
import strategy.fee.PartnershipFee

class StrategyTest : FunSpec({
    test("strategy run") {
        val amount = 1000.0
        val defaultFee = amount * Fee.DEFAULT_FEE
        val partnershipFee = amount * Fee.PARTNERSHIP_FEE

        val abcBankRemit = AbcBankRemit(DefaultFee())
        val defBankRemit = DefBankRemit(PartnershipFee())

        val amountAbc = abcBankRemit.calculateAmount(amount)
        val amountDef = defBankRemit.calculateAmount(amount)

        amountAbc shouldBe amount + defaultFee
        amountDef shouldBe amount + partnershipFee
    }
})